crypto/ffc/ffc_params_validate.c

```diff
@@ -135,9 +139,14 @@ int ossl_ffc_params_simple_validate(OSSL_LIB_CTX *libctx, const FFC_PARAMS *para
 int ossl_ffc_params_full_validate(OSSL_LIB_CTX *libctx, const FFC_PARAMS *params,
                                   int paramstype, int *res)
 {
+    int tmpres = 0;
+
     if (params == NULL)
         return 0;
 
+    if (res == NULL)
+        res = &tmpres;
+
```

Comment:
maybe it is clearer just to do
````
#ifdef FIPS_MODULE
        return ossl_ffc_params_FIPS186_4_validate(libctx, params, paramstype, res, NULL);
#else
     if ( )
        return 
    ....
#endif
````

Reply:
OK, done, although that duplicates the code a little. But the ifdefs are more clear this way.